flowmachine/flowmachine/features/subscriber/subscriber_event_count.py

```diff
@@ -0,0 +1,126 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# -*- coding: utf-8 -*-
+
+import warnings
+
+from ...core import Table
+from ...core.errors.flowmachine_errors import MissingDirectionColumnError
+from ..utilities.sets import EventsTablesUnion
+from .metaclasses import SubscriberFeature
+
+
+class SubscriberEventCount(SubscriberFeature):
+    """
+    This class returns the event count per subscriber within the period,
+    optionally limited to only incoming or outgoing events.
+
+    Parameters
+    ----------
+    start, stop : str
+         iso-format start and stop datetimes
+    hours : 2-tuple of floats, default 'all'
+        Restrict the analysis to only a certain set
+        of hours within each day.
+    subscriber_identifier : {'msisdn', 'imei'}, default 'msisdn'
+        Either msisdn, or imei, the column that identifies the subscriber.
+    subscriber_subset : str, list, flowmachine.core.Query, flowmachine.core.Table, default None
+        If provided, string or list of string which are msisdn or imeis to limit
+        results to; or, a query or table which has a column with a name matching
+        subscriber_identifier (typically, msisdn), to limit results to.
+    direction : {'in', 'out', 'both'}, default 'out'
+        Whether to consider calls made, received, or both. Defaults to 'out'.
+    tables : str or list of strings, default 'all'
+        Can be a string of a single table (with the schema)
+        or a list of these. The keyword all is to select all
+        subscriber tables
+
+    Examples
+    --------
+
+    >>> s = SubscriberEventCount("2016-01-01", "2016-01-07", direction="in")
+    >>> s.get_dataframe()
+
+             subscriber  event_count
+    0  2ZdMowMXoyMByY07           65
+    1  MobnrVMDK24wPRzB           81
+    2  0Ze1l70j0LNgyY4w           57
+    3  Nnlqka1oevEMvVrm           63
+    4  4dqenN2oQZExwEK2           59
+                    ...          ...
+    """
+
+    def __init__(
+        self,
+        start,
+        stop,
+        *,
+        subscriber_identifier="msisdn",
+        direction="both",
+        hours="all",
+        subscriber_subset=None,
+        tables="all",
+    ):
+        self.start = start
+        self.stop = stop
+        self.subscriber_identifier = subscriber_identifier
+        self.direction = direction
+        self.hours = hours
+
+        if direction not in {"in", "out", "both"}:
+            raise ValueError("{} is not a valid direction.".format(self.direction))
+
+        if self.direction == "both":
+            column_list = [self.subscriber_identifier]
+            self.tables = tables
+        else:
+            column_list = [self.subscriber_identifier, "outgoing"]
+            self.tables = self._parse_tables_with_direction(tables)
```

Comment:
Very nice, this reads much clearer now that you've moved the parsing code into a helper method.

Tiny suggestion: the name `_parse_tables_with_direction` seems to imply that it only parses the tables that have a direction and omits the others, when in fact it raises an error in this case (as it should). How about renaming it to `_parse_tables_ensuring_direction_is_present` or something similar?